bin/install

```diff
@@ -6,6 +6,16 @@ set -q OMF_REPO_URI;    or set OMF_REPO_URI "https://github.com/oh-my-fish/oh-my
 set -q OMF_REPO_BRANCH; or set OMF_REPO_BRANCH "master"
 
 
+# fish_exit polyfill for Fish versions < 3.
+if not set -q fish_pid
+  eval '
+    function fish_exit_polyfill --on-process-exit %self
+        emit fish_exit $argv[3]
+    end
+  '
+end
```

Comment:
How about ditching the super slow `eval` and leaving only the function declaration itself:
```fish
if not set -q fish_pid
  function fish_exit_polyfill --on-process-exit %self
    emit fish_exit $argv[3]
  end
end
```
And we get increased readability as a bonus 😉

Reply:
Well if %-substitution is deprecated, then when it is removed will the above yield a syntax error when the file is parsed? Or will it just be a string whose value is `"%self"`? I originally used `eval` so that the deprecated syntax is conditionally parsed.

Reply:
replace `%self` with `%monkey` to test :) 

the `if not set` seems to prevent the syntax error:
<img width="762" alt="screen shot 2019-01-07 at 8 20 39 am" src="https://user-images.githubusercontent.com/53660/50779435-255eca00-1255-11e9-9a25-06c1d325f13c.png">

Reply:
1. %self will stay for at least a while

2. There is no *syntax error* here - without process expansion, the `%self` would simply stay `%self`, and the option to `function` will only be evaluated once that's actually called, so even if `--on-process-exit` was removed it would still work if it was never called.

So yeah, there's no need for the `eval`.